src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2445,23 +2445,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
         return false;
       }
       break;
-    // Some types of VectorCast are not implemented for now.
-    case Op_VectorCastI2X:
-      if (bt == T_BYTE) {
-        return false;
-      }
-      break;
     case Op_VectorCastS2X:
       if (vlen < 4 || bit_size < 64) {
         return false;
       }
       break;
```

Comment:
Why do you remove others but keep this (4Sto4B not implemented)?

Reply:
Because @theRealELiu suggest us not to have VectorCast with bit_size less than 64 bits support.
In addition, we will add rules for the 4B vector, which will appear in actual situations. I'll refine the code in next commit.

Reply:
Byte is special here. 

Refs:
[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/matcher.hpp#L341
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/aarch64.ad#L2491

Reply:
We have removed it.